src/skills/Attack.cpp

```diff
@@ -19,7 +19,14 @@ namespace ai {
 Attack::Attack(string name, bt::Blackboard::Ptr blackboard)
         :Skill(std::move(name), std::move(blackboard)) {
 }
+Attack::Attack(string name, bt::Blackboard::Ptr blackboard, rtt::ai::coach::OffensiveCoach offensiveCoach)
+        : Skill(std::move(name), std::move(blackboard)) {
+    this->offensiveCoach = offensiveCoach;
+}
 
+rtt::ai::coach::OffensiveCoach Attack::getOffensiveCoach() {
+    return this->offensiveCoach;
```

Comment:
Same thing; creating local coaches is never going to work since they coordinate global behavior. All the skills need to be referring to the same object for coaches to work.